pkg/manager/manager.go

```diff
@@ -22,6 +27,18 @@ import (
 	"github.com/devtools-qe-incubator/eventmanager/pkg/util/logging"
 )
 
+const (
+	DefaultPipelineRunSyncInterval = 5 * time.Second
+)
+
+type BareMetalMachineConfig struct {
```

Comment:
I guess it is too early but this logic (hard part of the logic) should be on its own package...the manager should kept as much clean as possible

Reply:
I apologize for the messy state of this PR. It's still in a poc-ish state.

Let me at least move this portion to a separate package.

Reply:
yeah yean no need to apologies, I saw the WIP you can keep it as much messy as you want till it works and you start to cleanup it up.

Just went through it and added the comment. Feel free to manage it whenever you want not right now

Reply:
I've reorganized the code into a separate package named `autoscheduling` (not 100% sure about the naming)
```shell
pkg/manager/autoscheduling
├── baremetal.go                      # Handles connection and status retrieval for QE lab instances  
└── pipelinerun_manager.go            # Manages status updates for Pending/Completed PipelineRuns  
```
Let me know your thoughts on the package structure and naming!

Reply:
Yeah that seems perfect, may I even reduce the name to `scheduling` or `scheduler` and then inside the type of resource scheduled in this case `pipelines` but totally up to you.